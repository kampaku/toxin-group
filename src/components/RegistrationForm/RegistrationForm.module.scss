@import 'scss/utils/font-template.scss';
@import 'scss/utils/vars.scss';

.form {
  max-width: 380px;
  width: 100%;
  padding: 40px 30px 30px;
  background-color: $colorBackground;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-shadow: 0px 0px 25px rgba(0, 0, 0, 0.2);
  border-radius: 4px;

  @media (max-width: $widthPhone) {
    padding: 30px 20px 20px;
  }
}

.title {
  @extend %sub-section-title;
  margin: 0 0 20px;
}

.label {
  @extend %label-text;
  margin-bottom: 5px;
}

.inputWrapper {
  display: flex;
  flex-direction: column;
}

.buttonWrapper {
  margin-bottom: 30px;
}

.login {
  display: grid;
  align-items: center;
  grid-template-columns: auto 99px;
  column-gap: 15px;
  height: 44px;
}

.text {
  @extend %body-text;
}

.wrapper {
  display: flex;
  flex-direction: column;
  row-gap: 10px;

  &_xl {
    margin-bottom: 24px;
  }

  &_m {
    margin-bottom: 22px;
  }
}
