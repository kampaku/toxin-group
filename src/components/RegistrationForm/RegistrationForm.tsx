import { FC, FormEvent, useState } from 'react';
import cn from 'classnames';
import { useTranslation } from 'next-i18next';

import { Input, RadioButtonGroup, Toggle, Button, MaskInput } from 'components';

import styles from './RegistrationForm.module.scss';

type Props = {
  handleOnSubmit: (
    email: string,
    password: string,
    data: RegistrationData
  ) => void;
};

const RegistrationForm: FC<Props> = ({ handleOnSubmit }) => {
  const [email, setEmail] = useState('');
  const handleInputEmailOnChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setEmail(e.target.value);
  };
  const { t } = useTranslation('registration-form');

  const [password, setPassword] = useState('');
  const handleInputPasswordOnChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setPassword(e.target.value);
  };
  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    const formData = new FormData(e.currentTarget as HTMLFormElement);
    const data: RegistrationData = Object.fromEntries(
      formData
    ) as RegistrationData;
    handleOnSubmit(email, password, data);
  };

  return (
    <form className={styles.form} onSubmit={handleSubmit}>
      <h3 className={styles.title}>{t('title')}</h3>
      <div className={cn(styles.wrapper, styles.wrapper_xl)}>
        <Input placeholder={t('name')} name="name" />
        <Input placeholder={t('surname')} name="surname" />
        <RadioButtonGroup
          buttonsProps={[
            { name: 'gender', text: t('male'), value: 'male' },
            { name: 'gender', text: t('female'), value: 'female' },
          ]}
          active="male"
        />
      </div>
      <div className={cn(styles.wrapper, styles.wrapper_m)}>
        <div className={styles.inputWrapper}>
          <span className={styles.label}>{t('birthday')}</span>
          <MaskInput name="birthday" />
        </div>
      </div>
      <div className={cn(styles.wrapper, styles.wrapper_m)}>
        <div className={styles.inputWrapper}>
          <span className={styles.label}>{t('loginInformation')}</span>
          <Input
            placeholder="Email"
            type="email"
            onChange={handleInputEmailOnChange}
          />
        </div>
        <Input
          placeholder={t('password')}
          type="password"
          onChange={handleInputPasswordOnChange}
        />
        <Toggle text={t('specOffers')} name="getSpecOffers" />
      </div>
      <div className={styles.buttonWrapper}>
        <Button type="submit">{t('buttonSignUp')}</Button>
      </div>
      <div className={styles.login}>
        <span className={styles.text}>{t('login')}</span>
        <Button type="signIn" href="/sign-in">
          {t('buttonSignIn')}
        </Button>
      </div>
    </form>
  );
};

export { RegistrationForm };
