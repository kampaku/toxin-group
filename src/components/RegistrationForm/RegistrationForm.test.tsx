import { screen, render, fireEvent } from '@testing-library/react';

import { RegistrationForm } from './RegistrationForm';

describe('Sign-in page', () => {
  let form: HTMLElement;
  const handleOnSubmit = jest.fn();

  beforeEach(() => {
    form = render(
      <RegistrationForm handleOnSubmit={handleOnSubmit} />
    ).container;
  });

  test('Snapshot', () => {
    expect(form).toMatchSnapshot();
  });

  test('Email input exist', () => {
    const emailInput = screen.getByPlaceholderText(/Email/i);
    expect(emailInput).toBeInTheDocument();
  });

  test('Email input work correctly', () => {
    const emailInput = screen.getByPlaceholderText(/Email/i);
    fireEvent.change(emailInput, { target: { value: 'test@gmail.com' } });
    expect(emailInput).toHaveValue('test@gmail.com');
  });

  test('Password input exist', () => {
    const passwordInput = screen.getByPlaceholderText(/Password/i);
    expect(passwordInput).toBeInTheDocument();
  });

  test('Password input work correctly', () => {
    const passwordInput = screen.getByPlaceholderText(/Password/i);
    fireEvent.change(passwordInput, { target: { value: '12345678' } });
    expect(passwordInput).toHaveValue('12345678');
  });

  test('Name input exist', () => {
    const nameInput = screen.queryByPlaceholderText('name');
    expect(nameInput).toBeInTheDocument();
  });

  test('Name input work correctly', () => {
    const nameInput = screen.getByPlaceholderText('name');
    fireEvent.change(nameInput, { target: { value: 'Test' } });
    expect(nameInput).toHaveValue('Test');
  });

  test('Surname input exist', () => {
    const surnameInput = screen.queryByPlaceholderText('surname');
    expect(surnameInput).toBeInTheDocument();
  });

  test('Surname input work correctly', () => {
    const surnameInput = screen.getByPlaceholderText('surname');
    fireEvent.change(surnameInput, { target: { value: 'Testing' } });
    expect(surnameInput).toHaveValue('Testing');
  });

  test('Gender radio group exist', () => {
    const male = screen.queryByDisplayValue('male');
    const female = screen.queryByDisplayValue('female');
    expect(male).toBeInTheDocument();
    expect(female).toBeInTheDocument();
  });

  test('Surname input work correctly', () => {
    const male = screen.getByDisplayValue('male');
    const female = screen.getByDisplayValue('female');
    expect(male).toBeChecked();
    fireEvent.click(female);
    expect(female).toBeChecked();
    fireEvent.click(male);
    expect(male).toBeChecked();
  });

  test('Birthday input exist', () => {
    const birthdayInput = screen.getByPlaceholderText('birthdayPlaceholder');
    expect(birthdayInput).toBeInTheDocument();
  });

  test('Birthday input work correctly', () => {
    const birthdayInput = screen.getByPlaceholderText('birthdayPlaceholder');
    fireEvent.change(birthdayInput, { target: { value: '' } });
    expect(birthdayInput).toHaveValue('');
    fireEvent.change(birthdayInput, { target: { value: '00' } });
    expect(birthdayInput).toHaveValue('00');
    fireEvent.change(birthdayInput, { target: { value: '5' } });
    expect(birthdayInput).toHaveValue('5');
  });

  test('Special offer input exist', () => {
    const specialOfferInput = screen.getByRole('checkbox');
    expect(specialOfferInput).toBeInTheDocument();
  });

  test('Special offer work correctly', () => {
    const specialOfferInput = screen.getByRole('checkbox');
    expect(specialOfferInput).not.toBeChecked();
    fireEvent.click(specialOfferInput);
    expect(specialOfferInput).toBeChecked();
    fireEvent.click(specialOfferInput);
    expect(specialOfferInput).not.toBeChecked();
  });

  test('Registration button exist', () => {
    const submitButton = screen.getByRole('button');
    expect(submitButton).toBeInTheDocument();
  });

  test('Sign In button exist', () => {
    const registrationButton = screen.getByRole('link');
    expect(registrationButton).toBeInTheDocument();
  });

  test('Sign in button can be called', () => {
    const submitButton = screen.getByRole('button');
    fireEvent.click(submitButton);
    expect(handleOnSubmit).toBeCalled();
  });
});
