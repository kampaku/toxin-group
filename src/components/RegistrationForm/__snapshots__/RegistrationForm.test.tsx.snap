// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in page Snapshot 1`] = `
<div>
  <form
    class="form"
  >
    <h3
      class="title"
    >
      title
    </h3>
    <div
      class="wrapper wrapper_xl"
    >
      <div
        class=""
      >
        <input
          class="input"
          name="name"
          placeholder="name"
          type="text"
          value=""
        />
      </div>
      <div
        class=""
      >
        <input
          class="input"
          name="surname"
          placeholder="surname"
          type="text"
          value=""
        />
      </div>
      <div
        class="radioButtonGroup"
      >
        <label
          class="radioButton"
        >
          <span
            class="radioButton__inputWrapper"
          >
            <input
              checked=""
              class="radioButton__control"
              name="gender"
              type="radio"
              value="male"
            />
            <span
              class="radioButton__border radioButton__border_active"
            />
          </span>
          <span
            class="radioButton__text radioButton__text_active"
          >
            male
          </span>
        </label>
        <label
          class="radioButton"
        >
          <span
            class="radioButton__inputWrapper"
          >
            <input
              class="radioButton__control"
              name="gender"
              type="radio"
              value="female"
            />
            <span
              class="radioButton__border"
            />
          </span>
          <span
            class="radioButton__text"
          >
            female
          </span>
        </label>
      </div>
    </div>
    <div
      class="wrapper wrapper_m"
    >
      <div
        class="inputWrapper"
      >
        <span
          class="label"
        >
          birthday
        </span>
        <input
          class="inputMask"
          name="birthday"
          placeholder="birthdayPlaceholder"
        />
      </div>
    </div>
    <div
      class="wrapper wrapper_m"
    >
      <div
        class="inputWrapper"
      >
        <span
          class="label"
        >
          loginInformation
        </span>
        <div
          class=""
        >
          <input
            class="input"
            name=""
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        class=""
      >
        <input
          class="input"
          name=""
          placeholder="password"
          type="password"
          value=""
        />
      </div>
      <label
        class="toggle"
      >
        <span
          class="toggle__inputWrapper"
        >
          <input
            class="toggle__control"
            name="getSpecOffers"
            type="checkbox"
          />
          <span
            class="toggle__border"
          />
        </span>
        <span
          class="toggle__text"
        >
          specOffers
        </span>
      </label>
    </div>
    <div
      class="buttonWrapper"
    >
      <button
        class="button button_type_submit"
        type="submit"
      >
        <span
          class="button__icon"
        >
          <svg
            fill="currentColor"
            height="24px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
        <span
          class="button_type_signInText"
        >
          buttonSignUp
        </span>
      </button>
    </div>
    <div
      class="login"
    >
      <span
        class="text"
      >
        login
      </span>
      <a
        class="button button_type_signIn"
        href="/sign-in"
      >
        <span
          class="button_type_signInText"
        >
          buttonSignIn
        </span>
      </a>
    </div>
  </form>
</div>
`;
